packages/alchemy/src/provider.ts

```diff
@@ -1,33 +1,35 @@
 import {
   BaseSmartContractAccount,
   SmartAccountProvider,
+  createPublicErc4337Client,
   deepHexlify,
   resolveProperties,
   type AccountMiddlewareFn,
   type SmartAccountProviderOpts,
 } from "@alchemy/aa-core";
-import type { Address, Chain, HttpTransport } from "viem";
+import { type Address, type Chain, type HttpTransport } from "viem";
 import {
   arbitrum,
   arbitrumGoerli,
   optimism,
   optimismGoerli,
 } from "viem/chains";
 import { SupportedChains } from "./chains.js";
+import type { ClientWithAlchemyMethods } from "./middleware/client.js";
+import { withAlchemyGasFeeEstimator } from "./middleware/gas-fees.js";
 import {
+  alchemyPaymasterAndDataMiddleware,
   withAlchemyGasManager,
   type AlchemyGasManagerConfig,
-  alchemyPaymasterAndDataMiddleware,
 } from "./middleware/gas-manager.js";
-import { withAlchemyGasFeeEstimator } from "./middleware/gas-fees.js";
-import type { ClientWithAlchemyMethods } from "./middleware/client.js";
 
 type ConnectionConfig =
   | {
       apiKey: string;
       rpcUrl?: undefined;
     }
-  | { rpcUrl: string; apiKey?: undefined };
+  | { rpcUrl: string; apiKey?: undefined }
+  | { rpcUrl: string; apiKey?: undefined; jwt: string };
```

Comment:
This is a breaking change, we need to support the case where RPC url is set but the other two are not set still.

Also nit: while we're here can we make the `?: undefined` -> `?: never`